cs-engine/1.12/index.md

```diff
@@ -1,14 +1,392 @@
+          ```
+
+5.  Configure `devicemapper`:
+
+    By default, the `devicemapper` graph driver does not come pre-configured in
+    a production-ready state. Follow the documented step by step instructions to
+    [configure devicemapper with direct-lvm for production](../../engine/userguide/storagedriver/device-mapper-driver/#/for-a-direct-lvm-mode-configuration)
+    to achieve the best performance and reliability for your environment.
+
+6.  Configure the Docker daemon to start automatically when the system starts,
+    and start it now.
+
+    ```bash
+    $ sudo systemctl enable docker.service
+    $ sudo systemctl start docker.service
+    ```
+
+7.  Confirm the Docker daemon is running:
+
+    ```bash
+    $ sudo docker info
+    ```
+
+8.  Only users with `sudo` access will be able to run `docker` commands.
+    Optionally, add non-sudo access to the Docker socket by adding your user
+    to the `docker` group.
+
+    ```bash
+    $ sudo usermod -a -G docker $USER
+    ```
+
+9. Log out and log back in to have your new permissions take effect.
+
+### Install on Ubuntu 14.04 LTS or 16.04 LTS
+
+1.  Install packages to allow `apt` to use a repository over HTTPS:
+
+    ```bash
+    $ sudo apt-get update
+
+    $ sudo apt-get install --no-install-recommends \
+        apt-transport-https \
+        curl \
+        software-properties-common
+    ```
+
+    Optionally, install additional kernel modules to add AUFS support.
+
+    ```bash
+    $ sudo apt-get install -y --no-install-recommends \
+        linux-image-extra-$(uname -r) \
+        linux-image-extra-virtual
+    ```
+
+2.  Download and import Docker's public key for CS packages:
+
+    ```bash
+    $ curl -fsSL 'https://sks-keyservers.net/pks/lookup?op=get&search=0xee6d536cf7dc86e2d7d56f59a178ac6c6238f52e' | sudo apt-key add -
+    ```
+
+3.  Add the repository. In the  command below, the `lsb_release -cs` sub-command
+    returns the name of your Ubuntu version, like `xenial` or `trusty`.
+
+    ```bash
+    $ sudo add-apt-repository \
+       "deb https://packages.docker.com/1.12/apt/repo/ \
+       ubuntu-$(lsb_release -cs) \
+       main"
+    ```
+
+4.  Install CS Docker Engine:
+
+    - **Latest version**:
+
+      ```bash
+      $ sudo apt-get update
+
+      $ sudo apt-get -y install docker-engine
+      ```
+
+    - **Specific version**:
+
+      On production systems, you should install a specific version rather than
+      relying on the latest.
+
+      1.  List the available versions:
+
+          ```bash
+          $ sudo apt-get update
+
+          $ apt-cache madison docker-engine
+          ```
+
+          The second column represents the version.
+
+      2.  Install a specific version by adding the version after `docker-engine`,
+          separeated by an equals sign (`=`):
+
+          ```bash
+          $ sudo apt-get install docker-engine=<version>
+          ```
+
+5. Confirm the Docker daemon is running:
```

Comment:
The snippet below is not rendering properly. I think you're missing a space here.